compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -283,7 +293,8 @@ OMR::X86::CodeGenerator::initialize(TR::Compilation *comp)
    // 32-bit platforms must check the processor and OS.
    // 64-bit platforms unconditionally support prefetching.
    //
-   if (_targetProcessorInfo.supportsSSE() && comp->target().cpu.testOSForSSESupport())
+   TR_ASSERT(comp->target().cpu.supportsFeature(OMR_FEATURE_X86_SSE) == _targetProcessorInfo.supportsSSE(), "supportsSSE() failed\n");
+   if (comp->target().cpu.supportsFeature(OMR_FEATURE_X86_SSE) && comp->target().cpu.testOSForSSESupport())
```

Comment:
Similar here, if we require OS support for SSE then the `supportsFeature` should return true if and only if the OS has support. We shouldn't keep checking twice every time. See [1] for example how this works on Z.

[1] https://github.com/eclipse/openj9/blob/91bd0c6d5e0cf4d094a7185cc6c70c80cc9269c7/runtime/port/unix/j9sysinfo.c#L963-L973

Reply:
`comp->target().cpu.testOSForSSESupport()` is a hardcoded true for OpenJ9 and false for OMR. There isn't an OS support check on x86.

```
bool
CPU::testOSForSSESupport()
   {
   return true; // VM guarantees SSE/SSE2 are available
   }
```

Can we just eliminate it and assume true for both OpenJ9 and OMR?

Reply:
In that case yes we should assume true then and fold the `testOSForSSESupport` API all together.